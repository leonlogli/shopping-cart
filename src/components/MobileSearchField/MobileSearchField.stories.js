import React from 'react'
import { action } from '@storybook/addon-actions'

import MobileSearchField from '.'

export default {
  title: 'MobileSearchField',
  parameters: { options: { selectedPanel: 'storybook/actions/panel' } },
}

export const mobileSearchField = () => (
  <MobileSearchField onSubmit={action('submit')} />
)
