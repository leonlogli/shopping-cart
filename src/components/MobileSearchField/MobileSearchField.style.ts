import styled from 'styled-components'
import Paper from '@material-ui/core/Paper'
import InputBase from '@material-ui/core/InputBase'
import MuiClearIcon from '@material-ui/icons/Clear'

const Input = styled(InputBase)`
  margin-left: 16px;
  flex: 1;
`

const MobileSearchPaneRoot = styled(Paper)`
  padding: 2px 4px;
  display: flex;
  align-items: center;
  width: 100%;
  min-height: 56px;
  /* border-bottom: 1px solid ${({ theme }) => theme.palette.divider}; */
`

const ClearIcon = styled(MuiClearIcon)`
  margin: 0 12px;
  cursor: pointer;
`

export default MobileSearchPaneRoot
export { MobileSearchPaneRoot, Input, ClearIcon }
