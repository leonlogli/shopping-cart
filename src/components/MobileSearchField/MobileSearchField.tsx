import React, {
  useState,
  FormEvent,
  ChangeEvent,
  MouseEvent,
  useEffect,
  forwardRef,
} from 'react'
import { InputBaseProps } from '@material-ui/core/InputBase'
import ArrowBackIcon from '@material-ui/icons/ArrowBack'
import IconButton from '@material-ui/core/IconButton'

import {
  MobileSearchPaneRoot,
  Input,
  ClearIcon,
} from './MobileSearchField.style'

export type MobileSearchFieldProps = InputBaseProps & {
  onSubmit?: (value?: string) => void
  /**
   * Callback invoked on back icon click
   */
  onBack?: (event?: MouseEvent<HTMLButtonElement>) => void
  value?: string
}

const MobileSearchField = forwardRef<HTMLDivElement, MobileSearchFieldProps>(
  (props, ref) => {
    const {
      className,
      id,
      placeholder = 'Search',
      onSubmit = () => null,
      onChange = () => null,
      inputProps,
      onBack,
      value = '',
      ...other
    } = props
    const [internalValue, setInternalValue] = useState(value)
    const [showClearButton, setShowClearButton] = useState(false)

    useEffect(() => {
      setInternalValue(value)
      if (internalValue) {
        setShowClearButton(true)
      }
    }, [value])

    const handleChange = (event: ChangeEvent<HTMLInputElement>) => {
      setInternalValue(event.target.value)
      setShowClearButton(!!event.target.value)
      onChange(event)
    }

    const handleClear = () => {
      setInternalValue('')
      setShowClearButton(false)
    }

    const handleFocus = () => {
      if (internalValue) {
        setShowClearButton(true)
      }
    }

    const handleSubmit = (event: FormEvent<HTMLDivElement>) => {
      event.preventDefault()
      onSubmit(internalValue)
    }

    return (
      <MobileSearchPaneRoot
        square
        component="form"
        onSubmit={handleSubmit}
        id={id}
        ref={ref}
        className={className}
        onFocus={handleFocus}
      >
        <IconButton aria-label="go back" color="inherit" onClick={onBack}>
          <ArrowBackIcon />
        </IconButton>
        <Input
          type="search"
          value={internalValue}
          onChange={handleChange}
          placeholder={placeholder}
          autoCapitalize="none"
          autoComplete="off"
          autoCorrect="off"
          spellCheck="false"
          inputProps={{ 'aria-label': placeholder, ...inputProps }}
          {...other}
        />
        {showClearButton && <ClearIcon onClick={handleClear} />}
      </MobileSearchPaneRoot>
    )
  }
)

export { MobileSearchField }
export default MobileSearchField
