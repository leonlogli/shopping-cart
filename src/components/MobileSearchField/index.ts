export { default } from './MobileSearchField'
export * from './MobileSearchField'
