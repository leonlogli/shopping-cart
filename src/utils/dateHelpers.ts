const durationFormat = (interval: number, type: string) => {
  return `il y a ${Math.floor(interval)} ${type}${
    Math.floor(interval) > 1 ? 's' : ''
  }`
}

const fromNow = (date: Date) => {
  const seconds = Math.floor(
    ((new Date() as any) - (new Date(date) as any)) / 1000
  )

  let interval = seconds / 31536000

  if (Math.floor(interval) >= 1) {
    return durationFormat(interval, 'an')
  }

  interval = seconds / 2592000
  if (Math.floor(interval) >= 1) {
    return `il y a ${Math.floor(interval)} mois`
  }

  interval = seconds / 86400
  if (Math.floor(interval) >= 1) {
    return durationFormat(interval, 'jour')
  }

  interval = seconds / 3600
  if (Math.floor(interval) >= 1) {
    return durationFormat(interval, 'heure')
  }

  interval = seconds / 60
  if (Math.floor(interval) >= 1) {
    return durationFormat(interval, 'minute')
  }

  return `Ã  l'instant`
}

/**
 * Parse the given string time to a date.Time must have `hh:mm:ss` format
 * @param time time to parse
 * @param date date to which the time will be set once parsed. If null, today's date is used
 */
const parseTimeToDate = (time: string, date?: Date) => {
  const dateToSet = date || new Date()
  const timeParts = time.split(':').map((t) => Number(t) || 0)

  dateToSet.setHours(timeParts[0], timeParts[1], timeParts[2] || 0)

  return dateToSet
}

const formatShortDate = (date: any, locale?: string) => {
  return new Date(date).toLocaleDateString(locale, {
    day: 'numeric',
    month: 'short',
    year: 'numeric',
  })
}

export { fromNow, parseTimeToDate, formatShortDate }
