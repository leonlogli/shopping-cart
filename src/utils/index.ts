export * from './testHelpers'
export * from './util'
export * from './dateHelpers'
