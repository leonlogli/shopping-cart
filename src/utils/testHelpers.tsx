import React from 'react'
import { ThemeProvider } from 'styled-components'
import { mount, shallow } from 'enzyme'
import { defaultTheme } from '../styles'

const ThemeProviderWrapper = ({ children }: any) => (
  <ThemeProvider theme={defaultTheme}>{children}</ThemeProvider>
)

export const shallowWithTheme = (tree: any) =>
  shallow(tree, {
    wrappingComponent: ThemeProviderWrapper,
  })

export const mountWithTheme = (tree: any) =>
  mount(tree, {
    wrappingComponent: ThemeProviderWrapper,
  })
