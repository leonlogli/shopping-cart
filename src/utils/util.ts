const toUSD = (value: number) => {
  return `${value.toLocaleString('en-US', {
    minimumFractionDigits: 2,
  })} USD`
}

export { toUSD }
