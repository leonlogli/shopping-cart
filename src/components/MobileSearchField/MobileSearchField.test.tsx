import React from 'react'

import MobileSearchField from './index'
import { mountWithTheme } from '../../utils'

describe('<MobileSearchField />', () => {
  const wrapper = mountWithTheme(<MobileSearchField />)

  it('should have the correct children', () => {
    expect(wrapper.find('form').length).toEqual(1)
  })

  it('should call the mock function when submitted', () => {
    const onSubmit = jest.fn()

    wrapper.setProps('onSubmit' as any, onSubmit)
    wrapper.simulate('submit')
    expect(onSubmit).toHaveBeenCalled()
  })

  it('should match a snapshot', () => {
    expect(wrapper).toMatchSnapshot()
  })
})
