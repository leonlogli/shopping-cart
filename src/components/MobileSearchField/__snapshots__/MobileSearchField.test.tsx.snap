// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<MobileSearchField /> should match a snapshot 1`] = `
<ForwardRef
  0="o"
  1="n"
  2="S"
  3="u"
  4="b"
  5="m"
  6="i"
  7="t"
>
  <Styled(WithStyles(ForwardRef(Paper)))
    component="form"
    onFocus={[Function]}
    onSubmit={[Function]}
    square={true}
  >
    <WithStyles(ForwardRef(Paper))
      className="gZhdDu"
      component="form"
      onFocus={[Function]}
      onSubmit={[Function]}
      square={true}
    >
      <ForwardRef(Paper)
        className="gZhdDu"
        classes={
          Object {
            "elevation0": "MuiPaper-elevation0",
            "elevation1": "MuiPaper-elevation1",
            "elevation10": "MuiPaper-elevation10",
            "elevation11": "MuiPaper-elevation11",
            "elevation12": "MuiPaper-elevation12",
            "elevation13": "MuiPaper-elevation13",
            "elevation14": "MuiPaper-elevation14",
            "elevation15": "MuiPaper-elevation15",
            "elevation16": "MuiPaper-elevation16",
            "elevation17": "MuiPaper-elevation17",
            "elevation18": "MuiPaper-elevation18",
            "elevation19": "MuiPaper-elevation19",
            "elevation2": "MuiPaper-elevation2",
            "elevation20": "MuiPaper-elevation20",
            "elevation21": "MuiPaper-elevation21",
            "elevation22": "MuiPaper-elevation22",
            "elevation23": "MuiPaper-elevation23",
            "elevation24": "MuiPaper-elevation24",
            "elevation3": "MuiPaper-elevation3",
            "elevation4": "MuiPaper-elevation4",
            "elevation5": "MuiPaper-elevation5",
            "elevation6": "MuiPaper-elevation6",
            "elevation7": "MuiPaper-elevation7",
            "elevation8": "MuiPaper-elevation8",
            "elevation9": "MuiPaper-elevation9",
            "outlined": "MuiPaper-outlined",
            "root": "MuiPaper-root",
            "rounded": "MuiPaper-rounded",
          }
        }
        component="form"
        onFocus={[Function]}
        onSubmit={[Function]}
        square={true}
      >
        <form
          className="MuiPaper-root gZhdDu MuiPaper-elevation1"
          onFocus={[Function]}
          onSubmit={[Function]}
        >
          <WithStyles(ForwardRef(IconButton))
            aria-label="go back"
            color="inherit"
          >
            <ForwardRef(IconButton)
              aria-label="go back"
              classes={
                Object {
                  "colorInherit": "MuiIconButton-colorInherit",
                  "colorPrimary": "MuiIconButton-colorPrimary",
                  "colorSecondary": "MuiIconButton-colorSecondary",
                  "disabled": "Mui-disabled",
                  "edgeEnd": "MuiIconButton-edgeEnd",
                  "edgeStart": "MuiIconButton-edgeStart",
                  "label": "MuiIconButton-label",
                  "root": "MuiIconButton-root",
                  "sizeSmall": "MuiIconButton-sizeSmall",
                }
              }
              color="inherit"
            >
              <WithStyles(ForwardRef(ButtonBase))
                aria-label="go back"
                centerRipple={true}
                className="MuiIconButton-root MuiIconButton-colorInherit"
                disabled={false}
                focusRipple={true}
              >
                <ForwardRef(ButtonBase)
                  aria-label="go back"
                  centerRipple={true}
                  className="MuiIconButton-root MuiIconButton-colorInherit"
                  classes={
                    Object {
                      "disabled": "Mui-disabled",
                      "focusVisible": "Mui-focusVisible",
                      "root": "MuiButtonBase-root",
                    }
                  }
                  disabled={false}
                  focusRipple={true}
                >
                  <button
                    aria-label="go back"
                    className="MuiButtonBase-root MuiIconButton-root MuiIconButton-colorInherit"
                    disabled={false}
                    onBlur={[Function]}
                    onDragLeave={[Function]}
                    onFocus={[Function]}
                    onKeyDown={[Function]}
                    onKeyUp={[Function]}
                    onMouseDown={[Function]}
                    onMouseLeave={[Function]}
                    onMouseUp={[Function]}
                    onTouchEnd={[Function]}
                    onTouchMove={[Function]}
                    onTouchStart={[Function]}
                    tabIndex={0}
                    type="button"
                  >
                    <span
                      className="MuiIconButton-label"
                    >
                      <ForwardRef>
                        <WithStyles(ForwardRef(SvgIcon))>
                          <ForwardRef(SvgIcon)
                            classes={
                              Object {
                                "colorAction": "MuiSvgIcon-colorAction",
                                "colorDisabled": "MuiSvgIcon-colorDisabled",
                                "colorError": "MuiSvgIcon-colorError",
                                "colorPrimary": "MuiSvgIcon-colorPrimary",
                                "colorSecondary": "MuiSvgIcon-colorSecondary",
                                "fontSizeInherit": "MuiSvgIcon-fontSizeInherit",
                                "fontSizeLarge": "MuiSvgIcon-fontSizeLarge",
                                "fontSizeSmall": "MuiSvgIcon-fontSizeSmall",
                                "root": "MuiSvgIcon-root",
                              }
                            }
                          >
                            <svg
                              aria-hidden={true}
                              className="MuiSvgIcon-root"
                              focusable="false"
                              viewBox="0 0 24 24"
                            >
                              <path
                                d="M20 11H7.83l5.59-5.59L12 4l-8 8 8 8 1.41-1.41L7.83 13H20v-2z"
                              />
                            </svg>
                          </ForwardRef(SvgIcon)>
                        </WithStyles(ForwardRef(SvgIcon))>
                      </ForwardRef>
                    </span>
                    <WithStyles(memo)
                      center={true}
                    >
                      <ForwardRef(TouchRipple)
                        center={true}
                        classes={
                          Object {
                            "child": "MuiTouchRipple-child",
                            "childLeaving": "MuiTouchRipple-childLeaving",
                            "childPulsate": "MuiTouchRipple-childPulsate",
                            "ripple": "MuiTouchRipple-ripple",
                            "ripplePulsate": "MuiTouchRipple-ripplePulsate",
                            "rippleVisible": "MuiTouchRipple-rippleVisible",
                            "root": "MuiTouchRipple-root",
                          }
                        }
                      >
                        <span
                          className="MuiTouchRipple-root"
                        >
                          <TransitionGroup
                            childFactory={[Function]}
                            component={null}
                            exit={true}
                          />
                        </span>
                      </ForwardRef(TouchRipple)>
                    </WithStyles(memo)>
                  </button>
                </ForwardRef(ButtonBase)>
              </WithStyles(ForwardRef(ButtonBase))>
            </ForwardRef(IconButton)>
          </WithStyles(ForwardRef(IconButton))>
          <Styled(WithStyles(ForwardRef(InputBase)))
            0="o"
            1="n"
            2="S"
            3="u"
            4="b"
            5="m"
            6="i"
            7="t"
            autoCapitalize="none"
            autoComplete="off"
            autoCorrect="off"
            inputProps={
              Object {
                "aria-label": "Search",
              }
            }
            onChange={[Function]}
            placeholder="Search"
            spellCheck="false"
            type="search"
            value=""
          >
            <WithStyles(ForwardRef(InputBase))
              0="o"
              1="n"
              2="S"
              3="u"
              4="b"
              5="m"
              6="i"
              7="t"
              autoCapitalize="none"
              autoComplete="off"
              autoCorrect="off"
              className="jokSNL"
              inputProps={
                Object {
                  "aria-label": "Search",
                }
              }
              onChange={[Function]}
              placeholder="Search"
              spellCheck="false"
              type="search"
              value=""
            >
              <ForwardRef(InputBase)
                0="o"
                1="n"
                2="S"
                3="u"
                4="b"
                5="m"
                6="i"
                7="t"
                autoCapitalize="none"
                autoComplete="off"
                autoCorrect="off"
                className="jokSNL"
                classes={
                  Object {
                    "adornedEnd": "MuiInputBase-adornedEnd",
                    "adornedStart": "MuiInputBase-adornedStart",
                    "colorSecondary": "MuiInputBase-colorSecondary",
                    "disabled": "Mui-disabled",
                    "error": "Mui-error",
                    "focused": "Mui-focused",
                    "formControl": "MuiInputBase-formControl",
                    "fullWidth": "MuiInputBase-fullWidth",
                    "input": "MuiInputBase-input",
                    "inputAdornedEnd": "MuiInputBase-inputAdornedEnd",
                    "inputAdornedStart": "MuiInputBase-inputAdornedStart",
                    "inputHiddenLabel": "MuiInputBase-inputHiddenLabel",
                    "inputMarginDense": "MuiInputBase-inputMarginDense",
                    "inputMultiline": "MuiInputBase-inputMultiline",
                    "inputTypeSearch": "MuiInputBase-inputTypeSearch",
                    "marginDense": "MuiInputBase-marginDense",
                    "multiline": "MuiInputBase-multiline",
                    "root": "MuiInputBase-root",
                  }
                }
                inputProps={
                  Object {
                    "aria-label": "Search",
                  }
                }
                onChange={[Function]}
                placeholder="Search"
                spellCheck="false"
                type="search"
                value=""
              >
                <div
                  0="o"
                  1="n"
                  2="S"
                  3="u"
                  4="b"
                  5="m"
                  6="i"
                  7="t"
                  autoCapitalize="none"
                  autoCorrect="off"
                  className="MuiInputBase-root jokSNL"
                  onClick={[Function]}
                  spellCheck="false"
                >
                  <input
                    aria-label="Search"
                    autoComplete="off"
                    className="MuiInputBase-input MuiInputBase-inputTypeSearch"
                    onAnimationStart={[Function]}
                    onBlur={[Function]}
                    onChange={[Function]}
                    onFocus={[Function]}
                    placeholder="Search"
                    type="search"
                    value=""
                  />
                </div>
              </ForwardRef(InputBase)>
            </WithStyles(ForwardRef(InputBase))>
          </Styled(WithStyles(ForwardRef(InputBase)))>
        </form>
      </ForwardRef(Paper)>
    </WithStyles(ForwardRef(Paper))>
  </Styled(WithStyles(ForwardRef(Paper)))>
</ForwardRef>
`;
